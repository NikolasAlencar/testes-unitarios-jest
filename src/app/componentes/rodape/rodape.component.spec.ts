import { ComponentFixture, TestBed } from '@angular/core/testing';
import { RodapeComponent } from './rodape.component';

describe('RodapeComponent', () => {
  let component: RodapeComponent;
  let fixture: ComponentFixture<RodapeComponent>;

  beforeEach(() => {
    TestBed.configureTestingModule({
      imports: [RodapeComponent],
    });

    fixture = TestBed.createComponent(RodapeComponent);
    component = fixture.componentInstance;
  });

  it('should be created', () => {
    expect(component).toBeTruthy();
  });

  it('should be defined inputs alt and src', () => {
    expect(component.alt).toBeDefined();
    expect(component.src).toBeDefined();
  });

  it('should render the content based in inputs alt and src', () => {
    component.src = 'Teste';
    component.alt = 'https://teste.com/teste.jpg';
    expect(component).toMatchSnapshot();
  });
});
