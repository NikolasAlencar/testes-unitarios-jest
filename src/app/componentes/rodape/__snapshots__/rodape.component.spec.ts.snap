// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RodapeComponent should render the content based in inputs alt and src 1`] = `
RodapeComponent {
  "__ngContext__": 4,
  "alt": "https://teste.com/teste.jpg",
  "src": "Teste",
}
`;
